packages/api/cms-api/src/user-permissions/auth/user-permissions.guard.ts

```diff
@@ -1,42 +1,65 @@
+import { RequiredPermission } from "../decorators/required-permission.decorator";
+import { ACCESS_CONTROL_SERVICE } from "../user-permissions.constants";
+import { AccessControlServiceInterface } from "../user-permissions.types";
 
 @Injectable()
 export class UserPermissionsGuard implements CanActivate {
-    constructor(protected reflector: Reflector, private readonly contentScopeService: ContentScopeService) {}
+    constructor(
+        protected reflector: Reflector,
+        private readonly contentScopeService: ContentScopeService,
+        @Inject(ACCESS_CONTROL_SERVICE) private readonly accessControlService: AccessControlServiceInterface,
+    ) {}
 
     async canActivate(context: ExecutionContext): Promise<boolean> {
         const isPublicApi = this.reflector.getAllAndOverride("publicApi", [context.getHandler(), context.getClass()]);
         if (isPublicApi) {
             return true;
         }
 
-        const scopeGuardActive = this.reflector.getAllAndOverride<ScopeGuardActiveMetadataValue | undefined>(SCOPE_GUARD_ACTIVE_METADATA_KEY, [
+        const request =
+            context.getType().toString() === "graphql" ? GqlExecutionContext.create(context).getContext().req : context.switchToHttp().getRequest();
+        const user = request.user as CurrentUserInterface | undefined;
+        if (!user) return false;
+
+        const requiredPermission = this.reflector.getAllAndOverride<RequiredPermission>("requiredPermission", [
             context.getHandler(),
             context.getClass(),
         ]);
+        if (!requiredPermission) {
+            throw new Error(`RequiredPermission decorator is missing in ${context.getClass().name}::${context.getHandler().name}()`);
+        }
 
-        if (scopeGuardActive === false) {
-            return true;
+        if (!this.isResolvingGraphQLField(context) && !requiredPermission.options?.skipScopeCheck) {
+            const contentScope = await this.contentScopeService.inferScopeFromExecutionContext(context);
+            if (!contentScope) {
+                throw new Error(
+                    `Could not get ContentScope. Either pass a scope-argument or add @AffectedEntity()-decorator or enable skipScopeCheck in @RequiredPermission() (${
+                        context.getClass().name
+                    }::${context.getHandler().name}())`,
+                );
+            }
+            if (!this.accessControlService.isAllowedContentScope(user, contentScope)) {
+                return false;
+            }
         }
 
-        const request =
-            context.getType().toString() === "graphql" ? GqlExecutionContext.create(context).getContext().req : context.switchToHttp().getRequest();
-        const user = request.user as CurrentUserInterface | undefined;
-        if (!user) return true;
+        return requiredPermission.requiredPermission.some((permission) => this.accessControlService.isAllowedPermission(user, permission));
```

Comment:
```suggestion
        return requiredPermission.requiredPermission.every((permission) => this.accessControlService.isAllowedPermission(user, permission));
```

Shouldn't this check if a user has all required permissions?